src/Utilities/copy.jl

```diff
@@ -575,29 +535,54 @@ only once all the model information is gathered.
 """
 function final_touch(::MOI.ModelLike, idxmap) end
 
+function default_copy_to(
+    dest::MOI.ModelLike,
+    src::MOI.ModelLike,
+    copy_names::Bool,
+    filter_constraints::Union{Nothing,Function} = nothing,
+)
+    @warn(
+        "The `copy_names` and `filter_constraints` arguments to " *
+        "`default_copy_to` are now keyword arguments.",
+        maxlog = 1,
+    )
+    return default_copy_to(
+        dest,
+        src;
+        copy_names = copy_names,
+        filter_constraints = filter_constraints,
+    )
+end
+
 """
     default_copy_to(
         dest::MOI.ModelLike,
-        src::MOI.ModelLike,
-        copy_names::Bool,
+        src::MOI.ModelLike;
+        copy_names::Bool = true,
         filter_constraints::Union{Nothing,Function} = nothing,
     )
 
-Implements `MOI.copy_to(dest, src)` by adding the variables and then the
-constraints and attributes incrementally. The function
-[`MathOptInterface.supports_incremental_interface`](@ref) can be used to check
-whether `dest` supports the copying a model incrementally.
+A default implementation of `MOI.copy_to(dest, src)` for models that implement
+the incremental interface, i.e., [`MOI.supports_incremental_interface`](@ref)
+returns `true`.
 
-If the `filter_constraints` arguments is given, only the constraints for which
-this function returns `true` will be copied. This function is given a
-constraint index as argument.
+If `filter_constraints` is a `Function`, only constraints for which
+`filter_constraints(ci)` returns `true` will be copied, where `ci` is the
+[`MOI.ConstraintIndex`](@ref) of the constraint.
 """
 function default_copy_to(
     dest::MOI.ModelLike,
-    src::MOI.ModelLike,
-    copy_names::Bool,
+    src::MOI.ModelLike;
+    copy_names::Bool = true,
     filter_constraints::Union{Nothing,Function} = nothing,
 )
+    if !MOI.supports_incremental_interface(dest, copy_names)
+        error(
```

Comment:
I'm wondering whether it makes sense to remove this error.
`MOI.supports_incremental_interface` is what is used by `instantiate`.
Would there be a case where you want instantiate to add a cache but you would still like to be able to call default_copy_to?
Removing the error is nonbreaking so it might be fine for now

Reply:
This check is trivial and will be in-lined. It also prevents a later error like "add variable not allowed" in some cases. I'd suggest we leave it in, at least until all the solvers are updated. We could remove it for 1.0 if it's not needed.

Reply:
> This check is trivial and will be in-lined

Yes, I was not worried about peformance

> I'd suggest we leave it in, at least until all the solvers are updated. We could remove it for 1.0 if it's not needed.

Sounds good